.github/actions/install-cli/action.yml

```diff
@@ -13,32 +13,34 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - name: yarn-build-with-cache
+    - name: pnpm-build-with-cache
       if: inputs.no-cache == 'false'
-      uses: ./.github/actions/yarn-build-with-cache
-      with:
-        ref: ${{ inputs.ref }}
+      uses: ./.github/actions/pnpm-build-with-cache
 
     - name: Setup Foundry
       if: inputs.no-cache != 'false'
       uses: ./.github/actions/setup-foundry
 
-    - name: yarn-install-build-without-cache
+    - name: Setup pnpm (no-cache path)
+      if: inputs.no-cache != 'false'
+      uses: pnpm/action-setup@v4
+
+    - name: pnpm-install-build-without-cache
       if: inputs.no-cache != 'false'
       shell: bash
       run: |
-        yarn install
-        yarn build
+        pnpm install --frozen-lockfile
+        pnpm build
 
     - name: Pack the CLI
       shell: bash
       id: pack-cli
       run: |
         cd typescript/cli
-        yarn pack
+        pnpm pack
```

Comment:
We could shorten the two commands to `pnpm -C typescript/cli pack`. But be sure to test where (which dir) the package ends up in, for the next step.